darts/tests/utils/test_model_selection.py

```diff
@@ -0,0 +1,104 @@
+from ..base_test_class import DartsBaseTestClass
+from darts.utils.model_selection import train_test_split
+from darts.utils.timeseries_generation import constant_timeseries
+
+
+def make_dataset(rows, cols):
+    return [constant_timeseries(i, cols) for i in range(rows)]
+
+
+def verify_shape(dataset, rows, cols):
+    return (
+        len(dataset) == rows and
+        all(len(row) == cols for row in dataset)
+    )
+
+
+class ClassTrainTestSplitTestCase(DartsBaseTestClass):
+    # test 1
+    # test parameters for axis = 0
+    def test_parameters_for_axis_0(self):
+        train_test_split(make_dataset(2, 10), axis=0, test_size=1)
+
+        # expecting no exception
+        self.assertTrue(True)
+
+    # test 2
+    # test parameteres for axis = 1
+    def test_parameters_for_axis_1_no_n(self):
+        with self.assertRaises(AttributeError):
+            train_test_split(make_dataset(1, 10), axis=1, horizon=1)
+
+    def test_parameters_for_axis_1_no_horizon(self):
+        with self.assertRaises(AttributeError):
```

Comment:
why should this fail? you have a default value for horizon.

Reply:
Since `n` and `horizon` are only valid parameters for `axis=1` they are set by default to 0 to cover the case of `axis=0`. But if the `axis=1` and any of these is set to 0, then it's an error.